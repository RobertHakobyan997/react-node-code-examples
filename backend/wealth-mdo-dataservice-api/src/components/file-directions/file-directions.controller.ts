import { Controller, Get } from '@nestjs/common';
import { ApiOkResponse } from '@nestjs/swagger';
import { Log } from 'ngpd-merceros-wealth-mdo-common-be/dist/logger/mdo-logger';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';

import { FileDirectionsService } from './file-directions.service';

@Controller('file-directions')
export class FileDirectionsController {
  constructor(private readonly service: FileDirectionsService) {
  }

  @ApiOkResponse({
    description: 'Returns all file directions',
    type       : [ Enum ],
  })
  @Get()
  @Log('Get supported files directions')
  findAll() {
    return this.service.findAll();
  }
}
