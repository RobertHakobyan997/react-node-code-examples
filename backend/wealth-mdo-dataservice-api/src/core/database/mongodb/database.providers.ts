import mongoose from 'mongoose';
import { Logger } from '@nestjs/common';
import { IS_LOCAL } from 'ngpd-merceros-wealth-mdo-common-be/dist/utils/environments';

import mongoConfig from '../../../config/mongo.config';

export const databaseProviders = [
  {
    // ToDo: change to dynamic provider and take it from mongoConfig
    provide   : 'DatabaseConnection',
    useFactory: (): Promise<typeof mongoose> => {
      if (IS_LOCAL()) {
        mongoose.set('debug', (collectionName, methodName, ...methodArgs) => {
          const logger = new Logger(collectionName);
          const args = methodArgs.reduce((acc, next, idx) => {
            acc += JSON.stringify(next, null, 2) + (idx === methodArgs.length - 1 ? '' : ', ');
            return acc;
          }, '');
          logger.log(`${methodName}(${args})`);
        });
      }
      return mongoose.connect(mongoConfig().uri, mongoConfig().options);
    },
  },
];
