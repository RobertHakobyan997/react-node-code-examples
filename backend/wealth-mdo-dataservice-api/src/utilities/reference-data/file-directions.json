[
  {
    "_id" : "ObjectId('608ba9cbc1656e7b68ab3ed5')",
    "key" : "inbound",
    "display" : "Inbound",
    "order" : 1,
    "hidden": false
  },
  {
    "_id" : "ObjectId('608ba9cbc1656e7b68ab3ed6')",
    "key" : "outbound",
    "display" : "Outbound",
    "order" : 2,
    "hidden": false
  },
  {
    "_id" : "ObjectId('608ba9cbc1656e7b68ab3ed7')",
    "key" : "error",
    "display" : "Error",
    "order" : 3,
    "hidden": true
  }
]
