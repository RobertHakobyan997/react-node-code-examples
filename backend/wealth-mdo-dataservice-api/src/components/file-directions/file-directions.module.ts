import { Module } from '@nestjs/common';
import FileDirectionsModel, { FileDirectionsModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-directions';

import { DatabaseModule } from '../../core/database/mongodb/database.module';

import { FileDirectionsController } from './file-directions.controller';
import { FileDirectionsService } from './file-directions.service';
import { FileDirectionsDal } from './file-directions.dal';

@Module({
  imports    : [ DatabaseModule ],
  controllers: [ FileDirectionsController ],
  providers  : [
    { provide: FileDirectionsModelToken, useFactory: () => FileDirectionsModel },
    FileDirectionsService,
    FileDirectionsDal
  ],
  exports: [ FileDirectionsService ],
})
export class FileDirectionsModule {
}
