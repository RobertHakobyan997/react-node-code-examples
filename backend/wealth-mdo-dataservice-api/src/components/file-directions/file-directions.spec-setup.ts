import { Test, TestingModule } from '@nestjs/testing';
import { FileDirectionsModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-directions';

import { mockMongooseModel } from '../../../test/mocks/mock-mongoose.model';

import { FileDirectionsService } from './file-directions.service';
import { FileDirectionsDal } from './file-directions.dal';

export const assert = async () => {
  const module: TestingModule = await Test.createTestingModule({
    providers: [
      FileDirectionsService,
      FileDirectionsDal,
      { provide: FileDirectionsModelToken, useValue: mockMongooseModel },
    ],
  }).compile();

  return module.get<FileDirectionsService>(FileDirectionsService);
};
