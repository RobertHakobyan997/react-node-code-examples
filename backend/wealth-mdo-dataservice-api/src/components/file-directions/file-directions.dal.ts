import { Inject, Injectable } from '@nestjs/common';
import { FilterQuery, Model, UpdateQuery } from 'mongoose';
import { Logger } from 'ngpd-merceros-logger-be';
import { plainToClass } from 'class-transformer';
import FileDirectionsModel, { FileDirectionsModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-directions';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';

@Injectable()
export class FileDirectionsDal {
  private readonly logger = new Logger(FileDirectionsDal.name);

  constructor(@Inject(FileDirectionsModelToken) private readonly model: Model<Document<Enum>>) {
  }

  findAll(filter: FilterQuery<Document<Enum>> = {}, sort: string | any = {}, options: any = {}) {
    return this.model
      .find(filter, {}, options)
      .sort(sort)
      .exec();
  }

  findOne(filter: FilterQuery<Document<Enum>> = {}) {
    return this.model.findOne(filter).exec();
  }

  findById(id: any, options: any = {}) {
    return this.model.findById(id, {}, options).exec();
  }

  insert(docs: any[]) {
    return this.model.insertMany(plainToClass(FileDirectionsModel, docs));
  }

  findByIdAndUpdate(id: any, docs: UpdateQuery<Enum>) {
    return this.model.findByIdAndUpdate(id, plainToClass(FileDirectionsModel, docs));
  }

  findByIdAndDelete(id: any) {
    return this.model.findByIdAndDelete(id);
  }
}
