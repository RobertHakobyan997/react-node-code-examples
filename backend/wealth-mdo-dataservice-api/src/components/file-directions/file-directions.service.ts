import { Injectable } from '@nestjs/common';
import { FilterQuery } from 'mongoose';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';

import { FileDirectionsDal } from './file-directions.dal';

@Injectable()
export class FileDirectionsService {
  constructor(private readonly dal: FileDirectionsDal) {
  }

  findAll(filter: FilterQuery<Document<Enum>> = {}, sort: string | any = {}, options: any = {}) {
    return this.dal.findAll(filter, sort, options);
  }
}
