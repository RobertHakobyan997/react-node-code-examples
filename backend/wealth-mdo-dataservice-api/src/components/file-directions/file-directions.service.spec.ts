import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { plainToClass } from 'class-transformer';
import FileDirectionsModel from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-directions';
import { Types } from 'mongoose';

import { fileDirections, fileDirectionsRaw } from '../../../test/data/file-directions.const';

import { FileDirectionsService } from './file-directions.service';
import { assert } from './file-directions.spec-setup';

describe('FileDirectionsService', () => {
  let service: FileDirectionsService;

  beforeEach(async () => {
    service = await assert();
  });

  it('plainToClass', () => {
    const mapped = plainToClass(FileDirectionsModel, fileDirectionsRaw);
    mapped.map(el => expect(el._id).toBeInstanceOf(Types.ObjectId));
    expect(mapped).toMatchSnapshot();
  });

  describe('getFileDirections()', () => {
    it('should return all file directions', async () => {
      jest.spyOn(service['dal'], 'findAll').mockResolvedValue(fileDirections as Document<Enum>[]);
      const res = await service.findAll();
      expect(res).toEqual(fileDirections);
      expect(res.length).toBe(3);
    });
  });
});
