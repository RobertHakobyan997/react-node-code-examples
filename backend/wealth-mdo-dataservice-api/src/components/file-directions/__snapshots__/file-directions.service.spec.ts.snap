// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDirectionsService plainToClass 1`] = `
Array [
  Object {
    "_id": "609d4c33bfd3901b8c3af7f9",
    "display": "Inbound",
    "hidden": false,
    "key": "inbound",
    "order": 1,
  },
  Object {
    "_id": "609d4c33bfd3901b8c3af7fa",
    "display": "Outbound",
    "hidden": false,
    "key": "outbound",
    "order": 2,
  },
  Object {
    "_id": "609d4c33bfd3901b8c3af7fb",
    "display": "Error",
    "hidden": false,
    "key": "error",
    "order": 3,
  },
]
`;
